internal/js_parser/js_parser.go

```diff
@@ -9508,8 +9508,10 @@ func (p *parser) isDotDefineMatch(expr js_ast.Expr, parts []string) bool {
 				return false
 			}
 
-			// The last symbol must be unbound
-			return p.symbols[result.ref.InnerIndex].Kind == js_ast.SymbolUnbound
+			p.ignoreUsage(result.ref)
```

Comment:
If we replace an injected symbol with a define, we need to explicitly mark it as unused (as `p.findSymbol` has the effect of marking usage). Otherwise, the injected import won't be eliminated.

Calling `p.ignoreUsage` feels a bit hacky. I also suppose that `ignoreUsage` should be invoked closer to the actual replacement callsite rather than in `isDotDefineMatch`, which doesn't necessitate that a replacement actually occurs...